services/slam/builtin/builtin.go

```diff
@@ -648,12 +664,14 @@ func (slamSvc *builtIn) StartSLAMProcess(ctx context.Context) error {
 	if slamSvc.port == localhost0 {
```

Comment:
[q] could this watcher for the port be instantiated in a separate go routine  before the slamProcess begins (up around line 650) and use channels to communicate the port value? That way there is no chance that that log is missed and keeps all logReader/Writer together? I know that the chances of the log being missed are vanishingly small but log parsing is a hairy task at best and wanted to bring this up as a possible option

Reply:
Good question! We'll actually read logs from the beginning because we generated the reader/writer pair before starting the process, and pass the writer as part of the process config.